src/tng-hooks.src.js

```diff
@@ -213,4 +224,45 @@
 			throw new Error("useEffect() only valid inside an Articulated Function or a Custom Hook.");
 		}
 	}
+
+	function useMemo(fn, ...guards) {
+		// passed in any guards?
+		if (guards.length > 0) {
+			// only passed a single guards list?
+			if (guards.length == 1 && Array.isArray(guards[0])) {
+				guards = guards[0];
+			}
+		}
+		// no guards passed
+		// NOTE: different handling than an empty guards list like []
+		else {
+			guards = undefined;
+		}
+
+		var bucket = getCurrentBucket();
+		if (bucket) {
+			// need to create this memoization-slot for this bucket?
+			if (!(bucket.nextMemoizationIdx in bucket.memoizations)) {
+				bucket.memoizations[bucket.nextMemoizationIdx] = [];
+			}
+
+			let memoizationIdx = bucket.nextMemoizationIdx;
+			let memoization = bucket.memoizations[memoizationIdx];
+
+			// check guards?
+			if (guardsChanged(memoization[1], guards)) {
+				// invoke the memoization
+				memoization[0] = fn();
+				memoization[1] = guards;
+			}
+
+			bucket.nextMemoizationIdx++;
+
+			return memoization[0];
```

Comment:
Note that this is where most of the changes from the main review message would go. This line 261 should make a return a function that checks if the value in the slot of `memoization[0]` had been called yet or not, invoke if it not, replace that slot with its return value, and then subsequently just keep returning that value from then on.